### DISCUSSION TITLE
plugin support release eta?

### DISCUSSION THREAD

**Discussion by leyloe** [helix-editor/helix#10984] [Q&A] [answered]
when will plugins be added? will it eventually be possible for a file tree, integrated terminal?

also final question, i got the debugger checkmark on hx --health for rust, but how do i use it? like placing debug breakpoints, and running in debug mode. i prefer to keep things vanilla and not use tmux

**Comment by David-Else** [comment, marked as answer]
> when will plugins be added?

This is a bad question, it annoys the maintainers, there is no set timetable.

> will it eventually be possible for a file tree, integrated terminal?

Yes, the file tree is already done and you can use it today if you want to test drive the plugin system in its alpha state: https://github.com/helix-editor/helix/pull/8675

> i got the debugger checkmark on hx --health for rust, but how do i use it? like placing debug breakpoints, and running in debug mode. i prefer to keep things vanilla and not use tmux

It is only a preview and not supported, you have to figure it out yourself my searching around this repo and the WIKI.

**Reply by leyloe** [reply]
> > when will plugins be added?
> 
> 
> 
> This is a bad question, it annoys the maintainers, there is no set timetable.
> 
> 
> 
> > will it eventually be possible for a file tree, integrated terminal?
> 
> 
> 
> Yes, the file tree is already done and you can use it today if you want to test drive the plugin system in its alpha state: https://github.com/helix-editor/helix/pull/8675
> 
> 
> 
> > i got the debugger checkmark on hx --health for rust, but how do i use it? like placing debug breakpoints, and running in debug mode. i prefer to keep things vanilla and not use tmux
> 
> 
> 
> It is only a preview and not supported, you have to figure it out yourself my searching around this repo and the WIKI. 
> 
> 

will the plugin system be an abi or a totally separate language

**Reply by TornaxO7** [reply]
@fu3x11 see https://github.com/helix-editor/helix/pull/8675 for more information

**Reply by batyrmuhadov** [reply]
> This is a bad question, it annoys the maintainers, there is no set timetable.

What does it mean it's a bad question? If you do something and offer it to people then be kind and do it without any complaining, or just don't do it. I am not complaining when I'm donating something or don't talk how it's hard to earn money

**Reply by David-Else** [reply]
Many people frequently ask when a plugin system will be available. This question has been answered numerous times, and repeatedly asking it suggests that the person has not researched this common query. Please note that this is my personal opinion and I do not represent the project.

**Reply by leyloe** [reply]
> Many people frequently ask when a plugin system will be available. This question has been answered numerous times, and repeatedly asking it suggests that the person has not researched this common query. Please note that this is my personal opinion and I do not represent the project.

bro what's your issue? just ignore the question without causing drama, it was a simple, valid question that i asked in a Q&A. Q&A's probably have duplicate questions on a daily basis anyway,this is meant to create answers fit for the user who asked.

**Reply by David-Else** [reply]
Sorry if I upset you, it was not my intention. Please note my original answer has thumbs up from maintainers. I am just trying to help out, I can assure you I don't have any issues with you or anyone else here. The word 'bad' was too strong, I apologize for that.

**Reply by kirawi** [reply]
> > Many people frequently ask when a plugin system will be available. This question has been answered numerous times, and repeatedly asking it suggests that the person has not researched this common query. Please note that this is my personal opinion and I do not represent the project.
> 
> bro what's your issue? just ignore the question without causing drama, it was a simple, valid question that i asked in a Q&A. Q&A's probably have duplicate questions on a daily basis anyway,this is meant to create answers fit for the user who asked.

We prefer having single issues or discussions dedicated to a particular topic rather than multiple ones covering the same thing. It fragments discussion and eats up maintainer bandwidth.

In addition to what @David-Else already said, I understand that it can feel frustrating, but at the end of the day we all want Helix to succeed, and it would help us out tremendously if you spend some time searching for previous threads before creating a new one.
